addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -86,3 +125,87 @@ def _should_show_transfers(self):
         if len(self.batch_id) == 1 and self == self.batch_id.picking_ids:
             return False
         return super()._should_show_transfers()
+
+    def _find_auto_batch(self):
+        # Search for other pickings that shares the same auto_batch criteria
+        domain = self._get_auto_batch_domain()
+        # By ordering pickings by their batch_id, we try to put the picking into an existing batch before trying to make a new one.
+        batchable_pickings = self.env['stock.picking'].search(domain, order="batch_id ASC")
+        full_batch_ids = set()
+        for picking in batchable_pickings:
+            if picking.batch_id and picking.batch_id.id not in full_batch_ids:
+                if picking.batch_id._is_picking_auto_mergeable(self):
+                    # Add picking to batch
+                    picking.batch_id.picking_ids |= self
+                    return picking.batch_id
+                else:
+                    full_batch_ids.add(picking.batch_id.id)
+            if not picking.batch_id and self._is_auto_batchable(picking):
+                # Create new batch with both pickings
+                new_batch = self.env['stock.picking.batch'].create({
+                    'picking_ids': [(4, self.id), (4, picking.id)],
+                    'company_id': self.company_id.id if self.company_id else False,
+                    'picking_type_id': self.picking_type_id.id,
+                })
+                if picking.picking_type_id.batch_auto_confirm:
+                    new_batch.action_confirm()
+                return new_batch
+
+        # If nothing came out of the loop, then no batch is doable given the conditions
+        return False
+
+    def _is_auto_batchable(self, picking):
+        """ Verifies if a picking can be put in a batch with another picking without violating auto_batch constrains.
+        """
+        res = True
+        if self.picking_type_id.batch_max_lines:
+            res = res and (len(self.move_ids) + len(picking.move_ids) <= self.picking_type_id.batch_max_lines)
+        if self.picking_type_id.batch_max_pickings:
+            # Sounds absurd. BUT if we restrict "batch max picking" to 1, makes sense ... Or not. Because then there is no point to batch.
```

Comment:
We can also enter a negative value :monocle_face: 
In those cases, the picking will not be grouped, I think it's not a big deal if the user enters dumb values like that and then it doesn't batch :slightly_smiling_face: